core/src/setup/Kokkos_Setup_SYCL.hpp

```diff
@@ -17,6 +17,10 @@
 #ifndef KOKKOS_SETUP_SYCL_HPP_
 #define KOKKOS_SETUP_SYCL_HPP_
 
+// This seems to be necessary for SYCL+Cuda and gives better performance
+// otherwise
+#define KOKKOS_IMPL_SYCL_USE_IN_ORDER_QUEUES
```

Comment:
Make sure you print whether we are using in order queues or not from `print_configuration`